prov/usnic/src/usdf_eq.c

```diff
@@ -194,11 +197,27 @@ static ssize_t usdf_eq_readerr(struct fid_eq *feq,
 		goto done;
 	}
 
-	ret = usdf_eq_read_event(eq, NULL, entry, sizeof(*entry), flags);
+	ret = usdf_eq_read_event(eq, NULL, &entry, sizeof(entry), flags);
+
+	/* read the user's setting for err_data. */
+	err_data = buffer->err_data;
+	err_data_size = buffer->err_data_size;
+
+	/* Copy the entry. */
+	*buffer = entry;
+
+	/* If version >= 1.5, we copy err_data to user's provided buffer and
+	 * since we overwritten the value above, we restore saved value here. */
+	api_version = eq->eq_fabric->fab_attr.api_version;
+	if (FI_VERSION_GE(api_version, FI_VERSION(1,5))) {
```

Comment:
It could use some clarification, but from reading the manpage I think there should also be a check against the `err_data_size != 0` here. Got that impression from this text in the man page: 

> For compatibility purposes, if err_data_size is 0 on input, or the fab-
       ric was opened with release < 1.5, err_data  will  be  set  to  a  data
       buffer  owned  by the provider.